docs/acs-private-apis.md

```diff
@@ -204,4 +221,93 @@ Get predicted values for a node.
 
 ### Predictions API
 
-TBD
+#### Get Predictions
+Get predicted values for a node.
+
+* Path: `nodes/{nodeId}/predictions`
+* Method: `GET`
+* Response:
+    * Status: 200
+    * Content-Type: `application/json`
+    * Entry object (wrapped with list/pagination information):
+        * `id` [string]
+        * `property` [string]
+        * `predictionValue` [value]
+        * `previousValue` [value]
+        * `confidenceLevel` [number]
+        * `modelId` [string]
+        * `reviewStatus` [UNREVIEWED|CONFIRMED|REJECTED]
+        * `updateType` [AUTOFILL|AUTOCORRECT]
+    * Sample
+```json
+{
+  "list": {
+    "pagination": {
+      "count": 1,
+      "hasMoreItems": false,
+      "totalItems": 1,
+      "skipCount": 0,
+      "maxItems": 100
+    },
+    "entries": [
+      {
+        "entry": {
+          "confidenceLevel": 0.97,
+          "modelId": "56785678-5678-5678-5678-567856785678",
+          "property": "cm:description",
+          "reviewStatus": "UNREVIEWED",
+          "id": "9bc52780-a41f-4c54-8527-80a41f2c54ac",
+          "previousValue": "Budget file for the web site redesign",
+          "predictionValue": "budget",
+          "updateType": "AUTOCORRECT"
+        }
+      }
+    ]
+  }
+}
+```
+
+#### Submit Predictions [Internal]
+Submit predicted values for a node. Note that this is not intended to be used by the UI.
+
+* Path: `nodes/{nodeId}/predictions`
+* Method: `POST`
+* Request:
+    * Content-Type: `application/json`
+    * Body:
+        * `id` [string]
+        * `property` [string]
+        * `predictionValue` [value]
+        * `confidenceLevel` [number]
+        * `modelId` [string]
+        * `reviewStatus` [UNREVIEWED|CONFIRMED|REJECTED]
+        * `updateType` [AUTOFILL|AUTOCORRECT]
+        * `predictionDateTime` [timestamp]
+    * Sample:
+```json
+{
+    "property": "cm:title",
+    "predictionValue": "Hello",
+    "confidenceLevel": 0.95,
+    "updateType": "AUTOCORRECT",
+    "modelId": "56785678-5678-5678-5678-567856785678",
+    "predictionDateTime":"2024-09-06T11:45:00.000+0000",
+    "reviewStatus": "UNREVIEWED"
+}
+```
+
+#### Review Prediction
+Review a predicted value for a node (not ready for use yet - see ACS-7949).
```

Comment:
@MichalKinas I tested this locally and get an issue `Cannot invoke \"String.toUpperCase()\"` since `parameters.getParameter(PARAM_REVIEW_STATUS)` always comes back `null` in `PredictionsEntityResource`. Support for predictions isn't a priority at the moment, but I wanted to document the work we'd done, so wondered if you remembered whether this feature was finished or not. I couldn't see any e2e tests for this API.

Reply:
I remember it worked when I was testing it but it was quite long time ago and I think it wasn't tested with HX part so not sure what is it's status now